Content.Shared/Hands/EntitySystems/SharedHandsSystem.Pickup.cs

```diff
@@ -0,0 +1,155 @@
+    {
+        if (!Resolve(uid, ref handsComp, false))
+            return false;
+
+        var hand = handsComp.ActiveHand;
+        if (handName != null && !handsComp.Hands.TryGetValue(handName, out hand))
+            return false;
+
+        if (hand == null)
+            return false;
+
+        return TryPickup(uid, entity, hand, checkActionBlocker, animateUser, handsComp, item);
+    }
+
+    public bool TryPickupAnyHand(EntityUid uid, EntityUid entity, bool checkActionBlocker = true, bool animateUser = false, SharedHandsComponent? handsComp = null, SharedItemComponent? item = null)
+    {
+        if (!Resolve(uid, ref handsComp, false))
+            return false;
+
+        if (!TryGetEmptyHand(uid, out var hand, handsComp))
+            return false;
+
+        return TryPickup(uid, entity, hand, checkActionBlocker, animateUser, handsComp, item);
+    }
+
+    public bool TryPickup(EntityUid uid, EntityUid entity, Hand hand, bool checkActionBlocker = true, bool animateUser = false, SharedHandsComponent? handsComp = null, SharedItemComponent? item = null)
+    {
+        if (!Resolve(uid, ref handsComp, false))
+            return false;
+
+        if (!Resolve(entity, ref item, false))
+            return false;
+
+        if (!CanPickup(uid, entity, hand, checkActionBlocker, handsComp, item))
+            return false;
+
+        // animation
+        var xform = Transform(uid);
+        var coordinateEntity = xform.ParentUid.IsValid() ? xform.ParentUid : uid;
+        var initialPosition = EntityCoordinates.FromMap(coordinateEntity, Transform(entity).MapPosition);
+
+        PickupAnimation(entity, initialPosition, xform.LocalPosition, animateUser ? null : uid);
+        DoPickup(uid, hand, entity, handsComp);
+
+        return true;
+    }
+
+    public bool CanPickupAnyHand(EntityUid uid, EntityUid entity, bool checkActionBlocker = true, SharedHandsComponent? handsComp = null, SharedItemComponent? item = null)
+    {
+        if (!Resolve(uid, ref handsComp, false))
+            return false;
+
+        if (!TryGetEmptyHand(uid, out var hand, handsComp))
+            return false;
+
+        return CanPickup(uid, entity, hand, checkActionBlocker, handsComp, item);
+    }
+
+    public bool CanPickup(EntityUid uid, EntityUid entity, Hand hand, bool checkActionBlocker = true, SharedHandsComponent? handsComp = null, SharedItemComponent? item = null)
+    {
+        if (!Resolve(uid, ref handsComp, false))
+            return false;
+
+        var handContainer = hand.Container;
+        if (handContainer == null || handContainer.ContainedEntity != null)
+            return false;
+
+        if (checkActionBlocker && !_actionBlocker.CanPickup(uid))
+            return false;
+
+        if (!Resolve(entity, ref item, false))
+            return false;
+
+        if (TryComp(entity, out PhysicsComponent? physics) && physics.BodyType == BodyType.Static)
+            return false;
```

Comment:
feels like these checks should happen before `CanPickup` actionblocker

Reply:
also its kind of weird that this exists in tandem with `ActionBlockerSystem.CanPickup`, might be confusing as to which people should be using

Reply:
So the event should definitely be after those checks, ill change that.

In terms of avoiding confusion between this and the action blocker, I think maybe the action blocker should probably just not exist? Unlike something like CanInteract, CanSpeak, etc, I don't think anything will ever have to check CanPickup or CanDrop if it isn't going to be going through hands system, so those might as well just be done inside of hands system?

So I'll just move them over

Reply:
sounds good